src/codegen.c

```diff
@@ -361,31 +361,36 @@ static void codegen_expr(CodegenContext *ctx, Node *expr) {
     }
 
     if (expr->binary.op == TK_LBRACK) {
-      codegen_expr(ctx, rhs);
+      // TODO: Just use lhs operand of subscript operator when right hand
+      // side is a compile-time-known zero value.
 
       IRInstruction *subs_lhs = NULL;
-      if (lhs->type->kind != TYPE_ARRAY && lhs->type->kind == TYPE_POINTER) {
+      if (!type_is_array(lhs->type) && !type_is_pointer(lhs->type))
         ERR("Subscript operator may only operate on arrays and pointers, which type %T is not", lhs->type);
-      }
+
       if (lhs->kind == NODE_VARIABLE_REFERENCE) {
-        // TODO: Handle local variable references, somehow. How can we tell if it's local/static?
-        subs_lhs = ir_static_reference(ctx, as_span(lhs->var->name));
+        IRInstruction *var = lhs->var->val.node->ir;
```

Comment:
Semantically this `var` variable should perhaps be called `var_decl`, because the `var->val.node` pointer of a NODE_VARIABLE_REFERENCE is the corresponding variable declaration. That’s just a minor thing tho.

Reply:
f1f876c1d6facb6629b6e21501baf982331181fc

Should be all fixed :)